### PR TITLE
WIP ENH add sklearn transformer

### PR DESCRIPTION
**Pull request by eickenberg**
EDIT by EO:
Solves #277

@edouardoyallon @edouardoyallon @edouardoyallon 

as requested

### CODE REVIEW

File: kymatio/utils.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    def fit(self, X=None, y=None):
+        # no fitting necessary
+        return self
+
+    def predict(self, X):
+        
+        X_reshaped = X.reshape((-1,) + self.signal_shape)
+        transformed = self.scattering(
+            torch.from_numpy(X_reshaped).to(self.scattering.device))
```
Comment (edouardoyallon) [inline kymatio/utils.py]: I'm adding the backend numpy asap to avoid using torch with sklearn

File: kymatio/utils.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        """Creates an object that is compatible with the scikit-learn API
+        and implements the .transform method.
+        
+        Parameters
+        ==========
+
+        S: an instance of Scattering1D, Scattering2D or Scattering3D
+            This instance is called by the transformer
+
+        signal_shape: tuple
```
Comment (edouardoyallon) [inline kymatio/utils.py]: tuple of ints

File: kymatio/utils.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            The shape of one sample. Scikit-learn convention is to work with
+            2D matrices only, of shape (n_samples, n_features). Data is
+            delivered in this way and has to be reshaped before transform
+
+        Output
```
Comment (edouardoyallon) [inline kymatio/utils.py]: there is a shift(tab) missing here

File: kymatio/utils.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        ======
+        Y: ndarray, shape (n_samples, n_scattering_features)
+
+        """
+
+        self.scattering = S
+        self.signal_shape = signal_shape
+
+    def fit(self, X=None, y=None):
```
Comment (edouardoyallon) [inline kymatio/utils.py]: is it necessary to add this empty function?

**Comment by edouardoyallon** [comment]
FYI codecov/jenkins/travis aren't running here. I don't know why.

**Comment by edouardoyallon** [comment]
@eickenberg where are we?

File: kymatio/utils.py (hunk 4 above)
Comment (lostanlen) [reply kymatio/utils.py]: yes, probably, if the `ScatteringTransformer` is part of a sklearn's `Pipeline`

File: kymatio/utils.py (hunk 2 above)
Comment (eickenberg) [reply kymatio/utils.py]: thx, fixed

File: kymatio/utils.py (hunk 3 above)
Comment (eickenberg) [reply kymatio/utils.py]: where exactly? I don't see

File: kymatio/utils.py (hunk 4 above)
Comment (eickenberg) [reply kymatio/utils.py]: yeah, is part of the sklearn API

File: kymatio/utils.py (hunk 1 above)
Comment (eickenberg) [reply kymatio/utils.py]: alright, we'll have to modify this here whenever that is done

**Comment by eickenberg** [comment]
Missing as of now: an example

File: kymatio/utils.py (hunk 1 above)
Comment (edouardoyallon) [reply kymatio/utils.py]: that's done

**Comment by janden** [comment]
So the reason the tests are failing seems to be that your new `test_utils.py` has the same name as another test script in `kymatio/scattering1d/tests`, which is apparently not ok. Can you change the name?

**Comment by edouardoyallon** [comment]
@eickenberg given the np backend, could you redo this PR?

**Comment by eugenium** [comment]
closing and opening new one due to some forced push issues

**Pull request closed without merging**